packages/ngtools/webpack/src/transformers/ctor-parameters_spec.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { tags } from '@angular-devkit/core'; // tslint:disable-line:no-implicit-dependencies
+import { createTypescriptContext, transformTypescript } from './ast_helpers';
+import { downlevelConstructorParameters } from './ctor-parameters';
+
+function transform(input: string, additionalFiles?: Record<string, string>) {
+  const { program, compilerHost } = createTypescriptContext(input, additionalFiles);
+  const transformer = downlevelConstructorParameters(() => program.getTypeChecker());
+  const result = transformTypescript(undefined, [transformer], program, compilerHost);
+
+  return result;
+}
+
+describe('Constructor Parameter Transformer', () => {
+  it('supports classes in same module', () => {
+    const input = tags.stripIndent`
+      export class ClassInject {};
+
+      export class MyService {
+        constructor(v: ClassInject) {}
+      }
+    `;
+
+    const output = `export class ClassInject { } ; export class MyService { constructor(v) { } } MyService.ctorParameters = () => [ { type: ClassInject } ];`;
+
+    const result = transform(input);
+
+    expect(tags.oneLine`${result}`).toEqual(tags.oneLine`${output}`);
+  });
+
+  it('supports Inject decorators with interfaces in same module', () => {
+    const input = tags.stripIndent`
+      export interface InterInject {};
+      export const INTERFACE_INJECT = new InjectionToken<InterInject>('interface-inject');
+
+      export class MyService {
+        constructor(@Inject(INTERFACE_INJECT) v: InterInject) {}
+      }
+    `;
+
+    const output = `import * as tslib_1 from "tslib"; ; export const INTERFACE_INJECT = new InjectionToken('interface-inject'); let MyService = class MyService { constructor(v) { } }; MyService.ctorParameters = () => [ { type: undefined, decorators: [{ type: Inject, args: [INTERFACE_INJECT,] }] } ]; MyService = tslib_1.__decorate([ tslib_1.__param(0, Inject(INTERFACE_INJECT)) ], MyService); export { MyService };`;
```

Comment:
While its not a problem, it seems at the output contains 2 semi colons.  After the input of tsib